parlai/mturk/tasks/convai2_single_score_eval/worlds.py

```diff
@@ -0,0 +1,424 @@
+                if self.turn_idx >= self.n_turn:
+                    acts = [None]
+
+                    # General mark for this convo
+                    for idx, agent in enumerate(self.agents):
+                        control_msg['text'] = GMARK_MSG
+                        control_msg['general_mark_score'] = True
+                        agent.observe(validate(control_msg))
+                        acts[idx] = agent.act(timeout=self.max_resp_time)
+                        while acts[idx]['text'] not in self.ratings:
+                            control_msg['text'] = NAN_MSG
+                            agent.observe(validate(control_msg))
+                            acts[idx] = agent.act(timeout=self.max_resp_time)
+                        if 'text' in acts[idx] and \
+                                acts[idx]['text'] in self.ratings:
+                            self.gmark_score[idx] = int(acts[idx]['text'])
+
+                    # Good rounds selection
+                    for idx, agent in enumerate(self.agents):
+                        control_msg['text'] = GOOD_ROUND_MSG
+                        control_msg['good_rounds'] = True
+                        control_msg['rounds'] = '</ROUND>'.join([
+                            '\n'.join(
+                                [self.dialog[i][1], self.dialog[i + 1][1]])
+                            for i in range(0, len(self.dialog), 2)
+                        ])
+                        agent.observe(validate(control_msg))
+                        acts[idx] = agent.act(timeout=self.max_resp_time)
+                        if 'text' in acts[idx]:
+                            self.good_rounds.append(acts[idx])
+
+                    # Bad rounds selection
+                    for idx, agent in enumerate(self.agents):
+                        control_msg['text'] = BAD_ROUND_MSG
+                        control_msg['bad_rounds'] = True
+                        control_msg['rounds'] = '</ROUND>'.join([
+                            '\n'.join(
+                                [self.dialog[i][1], self.dialog[i + 1][1]])
+                            for i in range(0, len(self.dialog), 2)
+                        ])
+                        agent.observe(validate(control_msg))
+                        acts[idx] = agent.act(timeout=self.max_resp_time)
+                        if 'text' in acts[idx]:
+                            self.bad_rounds.append(acts[idx])
+
+                    self.chat_done = True
+                    for ag in self.agents:
+                        ag.observe(validate(acts[idx]))
+                        control_msg['text'] = CHAT_ENDED_MSG
+                        ag.observe(validate(control_msg))
+                return
+
+            self.dialog.append((idx, acts[idx]['text']))
+            if self.turn_idx == 1:
+                acts[idx]['text'] = self.model_persona_text + '\n' + \
+                    acts[idx]['text']
+            print(acts[idx])
```

Comment:
there are lots of prints in this file. some seem unnecessary (like this one) and perhaps others you'd rather log?